Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -274,22 +262,13 @@ extension BrowserViewController: URLBarDelegate {
     func urlBarDidPressReaderMode(urlBar: URLBarView) {
```

Comment:
It looks like you're adding selectedTab checks for navigation below, so we should move these to tab creation rather than registering/unregistering them when the tab changes.